PhotoMap/Features/Timeline/Controllers/TimelineViewController.swift

```diff
@@ -10,4 +10,58 @@ import Combine
 
 class TimelineViewController: BaseViewController {
     
+    // MARK: - Variables
+    private struct Constants {
+        static let cellIdentifier = "markerCell"
+    }
+    private var viewModel: TimelineViewModelType?
+    
+    // MARK: - UI Properties
+    @IBOutlet private weak var tableView: UITableView!
+    private let searchBar = UISearchBar(placeholder: L10n.Main.NavBar.Search.title)
+    private let categoryBarButton = UIBarButtonItem(title: L10n.Main.NavBar.Category.title, style: .plain, target: self, action: nil)
+    
+    // MARK: - Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+    }
+    
+    // MARK: - Helpers
+    static func newInstanse(viewModel: TimelineViewModelType) -> TimelineViewController {
+        let timelineVC = StoryboardScene.Timeline.timelineViewController.instantiate()
+        timelineVC.viewModel = viewModel
+        return timelineVC
+    }
+    
+    private func setupView() {
+        tableView.tableFooterView = UIView()
+        navigationItem.titleView = searchBar
+        navigationItem.rightBarButtonItem = categoryBarButton
+    }
+    
+}
+
+// MARK: - UITable view data source
+extension TimelineViewController: UITableViewDataSource {
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return viewModel?.numberOfSections ?? 0
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return viewModel?.headerTitles[section]
```

Comment:
Encapsulate logic (hide headerTitle array, use separated method to return title for section). View should know nothing about headerTitle array. It should be private in ViewModel and be accessible only through methods